ibllib/pipes/misc.py

```diff
@@ -1147,13 +1147,52 @@ class WindowsInhibitor:
     ES_CONTINUOUS = 0x80000000
     ES_SYSTEM_REQUIRED = 0x00000001
 
-    def __init__(self):
-        pass
+    @staticmethod
+    def _set_thread_execution_state(state: int) -> None:
+        result = ctypes.windll.kernel32.SetThreadExecutionState(state)
+        if result == 0:
+            log.error("Failed to set thread execution state.")
 
-    def inhibit(self):
-        print("Preventing Windows from going to sleep")
-        ctypes.windll.kernel32.SetThreadExecutionState(WindowsInhibitor.ES_CONTINUOUS | WindowsInhibitor.ES_SYSTEM_REQUIRED)
+    @staticmethod
+    def inhibit(quiet: bool = False):
+        if quiet:
+            log.debug("Preventing Windows from going to sleep")
+        else:
+            print("Preventing Windows from going to sleep")
+        WindowsInhibitor._set_thread_execution_state(WindowsInhibitor.ES_CONTINUOUS | WindowsInhibitor.ES_SYSTEM_REQUIRED)
+
+    @staticmethod
+    def uninhibit(quiet: bool = False):
+        if quiet:
+            log.debug("Allowing Windows from going to sleep")
+        else:
+            print("Allowing Windows to go to sleep")
+        WindowsInhibitor._set_thread_execution_state(WindowsInhibitor.ES_CONTINUOUS)
+
+
+def sleepless(func: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
Is there anyway to test the function (at least the syntax and arguments ? )
Even on linux you can probably decorate a dummy function and run it just to see that it returns the proper number of arguments and that the syntax is correct. You will at least get coverage from this.